__tests__/app.test.js

```diff
@@ -141,3 +141,41 @@ describe("/api/review/:review_id/comments", () => {
       });
   });
 });
+describe("POST a comment to given review", () => {
+  test("POST 201 responds with posted comment", () => {
+    const REVIEW_ID = 2;
+    return request(app)
+      .post(`/api/reviews/${REVIEW_ID}/comments`)
+      .send({ username: "bainesface", body: "hello" })
+      .expect(201)
+      .then((res) => {
+        expect(res.body.comment).toMatchObject({
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          review_id: REVIEW_ID,
+          author: expect.any(String),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+  test("404 username is invalid", () => {
+    const REVIEW_ID = 2;
+    return request(app)
+      .post(`/api/reviews/${REVIEW_ID}/comments`).send({ username:'robbie', body:'hello'})
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("invalid username");
+      });
+  });
+  test("404 comment body not found", () => {
+    const REVIEW_ID = 2;
+    return request(app)
+      .post(`/api/reviews/${REVIEW_ID}/comments`)
+      .send({ username: "bainesface", body: "" })
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("comment body not found");
+      });
+  });
+});
```

Comment:
What about if we post a comment to a review that doesnt exist? (9999)
or a review_id that isnt a number?
what if we send a comment that is missing some keys?